src/sagemaker/fw_utils.py

```diff
@@ -229,11 +270,25 @@ def create_image_uri(
         # 'cpu' or 'gpu'.
         if family in optimized_families:
             device_type = family
+        elif family.startswith("inf"):
+            device_type = "inf"
         elif family[0] in ["g", "p"]:
             device_type = "gpu"
         else:
             device_type = "cpu"
 
+    if device_type == "inf":
+        if region not in INFERENTIA_SUPPORTED_REGIONS:
+            raise ValueError("Inferentia is not supported in region {}.".format(region))
+        if framework not in INFERENTIA_VERSION_RANGES:
+            raise ValueError("Inferentia does not support {}.".format(framework.split("-")[-1]))
+        if not _is_inferentia_supported(framework, framework_version):
+            raise ValueError(
+                "Inferentia is not supported with {} version {}.".format(
+                    framework.split("-")[-1], framework_version
+                )
```

Comment:
It might be helpful to guide users on which regions, frameworks, and versions are supported in the error messages. For example, "Inferentia is not supported in region {}. The supported regions are {}." or something along those lines.